rundeckapp/grails-app/services/rundeck/services/ScmService.groovy

```diff
@@ -557,6 +570,42 @@ class ScmService {
         }
     }
 
+    /**
+     * It checks that the git base directory is not on any other git configuration
+     * @param project
+     * @param integration
+     * @param context
+     * @throws ScmPluginException
+     */
+    void validateIntegrationDirectory(project, integration,context) throws ScmPluginException {
+        ScmPluginConfigData currentPluginConfig = loadScmConfig(project, integration)
+        if(currentPluginConfig){
+            def currentScmPluginIntegration = loadPluginWithConfig(integration, context, currentPluginConfig.type, currentPluginConfig.getConfig(), false)
+            if(currentScmPluginIntegration && currentScmPluginIntegration.provider && currentScmPluginIntegration.provider.getBaseDirectoryPropertyValue()){
+                for(String projectName : frameworkService.getRundeckFramework().getFrameworkProjectMgr().listFrameworkProjectNames()){
```

Comment:
this is going to be very inefficient with a large number of projects.  can it just iterate on existing loadedPlugins?